azure-pipelines.yml

```diff
@@ -0,0 +1,72 @@
+strategy:
+  matrix:
+    UnitTest:
+      TESTKIND: "unit"
+    IntegrationTest:
+      TESTKIND: "integration"
+
+pool:
+  vmImage: 'vs2017-win2016'
+
+variables:
+  GOPATH: '$(system.defaultWorkingDirectory)\gopath' # Go workspace path
+  modulePath: '$(GOPATH)\src\github.com\$(build.repository.name)' # Path to the module's code
+  TERRAFORMVERSION: 0.11.13
+  LOGPARSERVERSION: v0.13.8
+
+steps:
+- pwsh: |
+    New-Item -ItemType "directory" -Path $(GOPATH)\pkg
+    New-Item -ItemType "directory" -Path $(modulePath)
+    Get-ChildItem -Path $(system.defaultWorkingDirectory) -Exclude gopath | Move-Item -Destination $(modulePath)
+    New-Item -ItemType "directory" -Path $(modulePath)\logs
+  displayName: 'Set up the Go workspace'
+
+- script: choco install dep
+  displayName: 'Install dep'
+
+- script: choco install terraform --version $(TERRAFORMVERSION)
+  displayName: 'Install Terraform'
+
+- script: dep ensure
+  workingDirectory: '$(modulePath)'
+  displayName: 'Run `dep ensure`'
+
+- pwsh: |
+    if ("$(TESTKIND)" -eq "unit") {
+      go test -v -parallel 1 $(go list ./... | where-object { $_ -notlike "*/test" }) | Tee-Object -FilePath 'logs/unit.log'
```

Comment:
I think we normally use `run-go-test` for this, which sets other options, such as `-timeout`. Can you make sure we re-use those here?

Also, perhaps a dumb question, but is there no way we can reuse any of those Bash scripts here? I'm curious if we can reduce duplication at all between Windows and *nix.

Reply:
I can run it on "Bash for Windows" (https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema?view=azure-devops&tabs=schema#bash). The problem with this approach is, the tests will still be running in the linux subsystem and not on Windows directly. That negates the whole point of setting up a Windows CI

Reply:
Ah, OK, fair enough

Reply:
Should we use python for our CI scripts once we have windows CI on track, to avoid the code duplication?

Reply:
Yea, probably. I mean, that still requires Python to be installed, and there will still be other OS differences to paper over (e.g., path lengths), but that should help.

Reply:
At least it isn't "share no code" 😄

Reply:
Why not Golang?  Installing Python is going to be a mess, someone's bound to install something and complain...